### PR TITLE
mysql-cluster: Fix build for Linuxbrew 

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

File: Formula/mysql-cluster.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ class MysqlCluster < Formula
 
   depends_on :java => "1.7+"
   depends_on "cmake" => :build
-  depends_on "pidof" unless MacOS.version >= :mountain_lion
+  depends_on "pidof" unless MacOS.version >= :mountain_lion && OS.mac?
```
Comment (rwhogg) [inline Formula/mysql-cluster.rb]: This needs to be `&& !OS.mac?`.

I remember I was frustrated by the double-negative of `unless` when I had to fix this for `mysql` too.

**Comment by iMichka** [comment]
This `unless` thing in ruby is a funny thing, but needs more focus when writing it. Not sure I like the idea :)

**Comment by rwhogg** [comment]
Ran out of memory with 16 threads. I've pushed to this branch to try 8.

**Comment by iMichka** [comment]
8 failed too. Pushed `-j4`.

**Comment by iMichka** [comment]
Yes that worked. Last failure, easy to fix:
> brew linkage --test mysql-cluster
Missing libraries:
  libncursesw.so.6

**Comment by sjackman** [comment]
Merged. Thanks, Michka!

**Comment by sjackman** [comment]
The bottle is 670 MB, which is too big to transfer to Bintray. The macOS bottle is 107 MB. Want to try stripping the executables and libraries?
https://bintray.com/homebrew/bottles/mysql-cluster#files

**Comment by iMichka** [comment]
Let's strip just the .a files in the lib folder. There is one .a file which is 500 Mb big ...

File: Formula/mysql-cluster.rb [hunk 2]
```diff
@@ -24,8 +24,11 @@ class MysqlCluster < Formula
 
   depends_on :java => "1.7+"
   depends_on "cmake" => :build
-  depends_on "pidof" unless MacOS.version >= :mountain_lion
+  if OS.mac?
+    depends_on "pidof" unless MacOS.version >= :mountain_lion
+  end
   depends_on "openssl"
+  depends_on "ncurses" unless OS.mac?
```
Comment (sjackman) [inline Formula/mysql-cluster.rb]:
```ruby
  if OS.mac?
    depends_on "pidof" unless MacOS.version >= :mountain_lion
  else
    depends_on "ncurses"
  end
````

**Comment by iMichka** [comment]
I did try to strip before, but it sort of failed because there was a mix of strippable/non-strippable files. But maybe my syntax was wrong. Let's try again

**Comment by iMichka** [comment]
```
==> strip /home/linuxbrew/.linuxbrew/Cellar/mysql-cluster/7.5.7/lib/libmysqld.a /home/linuxbrew/.linuxbrew/Cellar/mysql-cluster/7.5.7/lib/libmysqlclient.a /home/linuxbrew/.linuxbrew/Cellar/mysql-cluster/7.5.7/lib/libmysqlservices.a /home/linuxbrew/.linuxbrew/Cellar/mysql-cluster/7.5.7/lib/libndbclient_static.a

Error: Is a directory @ io_fread - /home/linuxbrew/.linuxbrew/Cellar/mysql-cluster/7.5.7/bin
```

**Comment by sjackman** [comment]
Try…
```ruby
    # Strip the binaries to reduce their size.
    unless OS.mac?
      system("strip", "--strip-unneeded", "--preserve-dates", *Dir[prefix/"**/*"].select do |f|
        f = Pathname.new(f)
        f.file? && (f.elf? || f.extname == ".a")
      end)
    end

**Comment by iMichka** [comment]
Nice, that worked. 135Mb instead of 500.

**Pull request closed without merging**